bazarr.py

```diff
@@ -79,19 +80,32 @@ def check_status():
     except FileNotFoundError:
         pass
 
+    running_processes=list()
+
+    def register_new_process(ep):
+        running_processes.append(ep)
+
+
+    def terminate_all_child_processes():
+        for ep in running_processes:
+            terminate_child_process(ep)
+
+    signal.signal(signal.SIGTERM, lambda signal_no, frame: terminate_all_child_processes())
+    atexit.register(terminate_all_child_processes)
+
     # Initial start of main bazarr process
     print("Bazarr starting...")
-    start_bazarr()
+    start_bazarr(register_new_process)
 
     # Keep the script running forever until stop is requested through term or keyboard interrupt
     while True:
-        check_status()
+        check_status(register_new_process)
         try:
             if sys.platform.startswith('win'):
                 time.sleep(5)
             else:
                 os.wait()
                 time.sleep(1)
-        except (KeyboardInterrupt, SystemExit):
+        except (KeyboardInterrupt, SystemExit, ChildProcessError):
```

Comment:
This error was raised whenever the `TERM` signal was sent to all child processes under Mac OS X.